### ISSUE TITLE
Proxy "use of closed network connection" error

### ISSUE THREAD

**Issue by fasaxc** [coreos/etcd#3107]
We see this intermittently.  We think a proxy gets disconnected from our (single) etcd cluster node for some reason (not sure why that happens).  It then gets the following error:
```
etcdserver: could not get cluster response from http://172.18.203.17:2380: Get http://172.18.203.17:2380/members: read tcp 172.18.203.17:2380: use of closed network connection
```
Then it seems like it only retries after a timeout rather than immediately.

At a guess, if there were multiple nodes in the cluster, it would retry another node and the connection would succeed straight away.

Not sure why the server would close the connection but maybe the proxy isn't properly handling that case so it appears as a read error on the next request?

**Comment by xiang90** [comment]
What etcd version are you running? I think we have fixed a related issue in the most recent released version.

**Comment by fasaxc** [comment]
Looks like 2.0.11.

**Comment by xiang90** [comment]
I think this has been fixed since 2.0.12 by #2902

**Comment by fasaxc** [comment]
@lukasa FYI, looks like that etcd issue has been fixed upstream
